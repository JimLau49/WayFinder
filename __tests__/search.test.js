import React from "react";
import Search from "../src/components/search/search.component";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("Search component", () => {
    it("should match snapshot", () => {
        const tree = renderer.create(<Search />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should call a function on user input", () => {
        const mockQueryText = jest.fn();
        const wrapper = shallow(
            <Search setDestination={mockQueryText} queryText={mockQueryText} />
        );
        const textInput = wrapper.find({testID: "searchInput"});
        textInput.simulate("changeText", 'H805');
        expect(mockQueryText).toHaveBeenCalledTimes(1);
        expect(mockQueryText.mock.calls[0][0]).toBe('H805');
    });
});
