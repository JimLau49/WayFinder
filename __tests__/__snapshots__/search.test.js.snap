// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "elevation": 5,
        "flex": 1,
        "height": 48,
        "position": "absolute",
        "top": 48,
        "width": 720,
        "zIndex": 1,
      },
      Object {
        "height": 48,
      },
    ]
  }
  testID="searchContainer"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#AA2B45",
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
        "paddingLeft": 16,
        "paddingRight": 16,
        "position": "relative",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={1}
        />
        <Image
          source={1}
          style={
            Object {
              "marginLeft": 16,
              "marginRight": 16,
            }
          }
        />
        <Image
          source={1}
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
          testID="searchInput"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <Image
        source={1}
      />
    </View>
  </View>
</View>
`;
